### DISCUSSION TITLE
LESSON 8: onlyOwner not withdrawing.

### DISCUSSION THREAD

**Discussion by olahamid** [Cyfrin/foundry-full-course-cu#901] [Q&A] [answered]
Hello,
Q1] I think I need more understanding with how the address are segmented.
the onlyowner address is the first address I used to connect, call the fund function and also use its private key in the make deploy. IS THIS STATEMENT CORRECT OR WRONG?

Q2]. The withdraw function doesn't withdraw any funds back and also i was able to call it the withdraw function with any other address, hence the onlyOwner isn't is working.

**Comment by 0xusmanf** [comment, marked as answer]
> Q1] I think I need more understanding with how the address are segmented. 
> the onlyowner address is the first address I used to connect, call the fund function and also use its private key in the make deploy. IS THIS STATEMENT CORRECT OR WRONG?

The onlyOwner address is the address you use to deploy the contract.

> Q2]. The withdraw function doesn't withdraw any funds back and also I was able to call it the withdraw function with any other address, hence the onlyOwner isn't is working.

Please share your code including the deploy scripts and make file, so we better understand the problem and guide you accordingly.

**Reply by olahamid** [reply]
Hello usman, 
Q1]. Well understood, thanks alot. 

Q2]. I didn't use --broadcast to deploy my script, did that and it worked just fine.

Big regards, thanks.👍
